javascript-loops/main.js

```diff
@@ -0,0 +1,73 @@
+/* exported getNumbersToTen,
+            getEvenNumbersToTwenty,
+            repeatWord,
+            logEachCharacter,
+            doubleAll,
+            getKeys,
+            getValues
+ */
+
+function getNumbersToTen() {
+  var numbers = [];
+  var currentNumber = 1;
+
+  while (currentNumber <= 10) {
+    numbers.push(currentNumber);
+    currentNumber++;
+  }
+
+  return numbers;
+}
+
+function getEvenNumbersToTwenty() {
+  var evenNumbers = [];
+  var currentNumber = 2;
+
+  while (currentNumber <= 20) {
+    evenNumbers.push(currentNumber);
+    currentNumber += 2;
+  }
+  return evenNumbers;
+}
+
+function repeatWord(word, time) {
+  var count = 1;
+  var repeated = '';
+
+  while (count <= time) {
+    repeated += word;
+    count++;
+  }
+  return repeated;
+}
+
+function logEachCharacter(string) {
+  for (var i = 0; i < string.length; i++) {
+    console.log(string[i]);
+  }
+}
+
+function doubleAll(numbers) {
+  var doubled = [];
+  for (var i = 0; i < numbers.length; i++) {
+    doubled[i] = numbers[i] * 2;
+  }
+  return doubled;
+}
+
+function getKeys(object) {
+  var keys = [];
+  for (var key in object) {
+    keys.push(key);
+  }
+  return keys;
+}
+
+function getValues(object) {
+  var values = [];
+  for (var key in object) {
+    values[key] = object[key];
```

Comment:
This is setting named properties in an array, which is a big no-no. You just want to push these values into that array.